decidim/decidim
# Soft delete for spaces and components

#### :tophat: What? Why?

This pull request implements the soft delete functionality for spaces, components, and resources within the administration panel. Instead of permanently deleting these elements, admins now have the option to send them to a recycle bin, allowing for restoration if needed. This feature enhances administrative flexibility.

#### :pushpin: Related Issues

- Fixes #13122

#### :hammer: How?

- [x] Added a `deleted_at` column to the relevant models.
- [x] Updated the `AddTimestampsToComponents` migration to use a temporary class within the migration. This change avoids errors caused by using the `Decidim::Component` model directly, which includes the SoftDeletable concern that relies on the `deleted_at` column.
- [x] Implemented soft delete methods and associated logic to handle the moving of items to the recycle bin:
   - [x] Participatory Processes
  - [x] Assemblies
  - [x] Conferences
  - [x] Components
  Resources:
    - [x] Proposals
    - [x] Accountability
    - [x] Budgets
    - [x] Blogs
    - [x] Debates
    - [x] Meetings
- [x] Updated the UI to include a trash can icon for delete actions.
- [x] Modified the global search to exclude deleted items.
- [x] Added specs to test the new functionality.

#### :white_check_mark: Testing

1. Go to the administration panel.
2. Attempt to delete a space or component.
3. Verify that the item moves to the recycle bin.
4. Check if the item can be restored from the recycle bin.
5. Ensure that deleted items do not appear in the global search.
6. Confirm that a notification is sent to resource authors upon deletion.

#### :camera: Screenshots

*Please add screenshots of the changes you are proposing*

![Description](URL)

:hearts: Thank you!
